Fixes #20083 - Search by environment for smart class parameter

orrabin opened the pull request:
Also moves seach by override to smart class parameter and renames search
by key in smart variable

ohadlevy commented on config/initializers/00_rest_client.rb:
```diff
@@ -0,0 +1,7 @@
+require 'rest_client'
```
left over? or maybe this should be redirected to its own logger (i would say thats a feature by it self).

ohadlevy commented on ponse):
```diff
@@ -0,0 +1,41 @@
+=> [[32m#<OVIRT::StorageDomain:0x00000004d79200[0m
```
not related, please delete.

orrabin committed:
Fixes #20083 - Search by environment for smart class parameter

Also moves seach by override to smart class parameter and renames search
by key in smart variable

iNecas commented:
Seems like the tests don't like the change for some reason…

orrabin commented:
@iNecas I was trying to refactor to get the search of override out of `variable lookup key` but there is only one united `lookup key` in api v1 so it broke.
I changed this to only add environment.

tbrisker commented on app/models/lookup_keys/puppetclass_lookup_key.rb:
```diff
@@ -9,6 +9,7 @@ class PuppetclassLookupKey < LookupKey
   after_validation :reset_override_params, :if => ->(key) { key.override_changed? && !key.override? }
 
   scoped_search :relation => :param_classes, :on => :name, :rename => :puppetclass, :aliases => [:puppetclass_name], :complete_value => true
+  scoped_search :relation => :environments, :on => :name, :rename => :environment, :complete_value => true
```
Should this (and possibly the previous one as well) be set to `:only_explicit`? I'm not sure if using free text to search for class or environment is common, and in a large setup this may have a possible performance effect.

iNecas replied:
:+1: on keeping the `:environments` `:only_explicit`. For the `param_classes`, it would mean a change in a behaviour, that is not related directly to the attached issue. I would probably keep it as separate issue, ideally opened by some user that would hit it.

orrabin commented:
@tbrisker I added `only_explicit`

tbrisker commented:
Thanks @orrabin !
